packages/headless/src/controllers/commerce/core/result-list/headless-core-interactive-result.ts

```diff
@@ -44,7 +57,19 @@ export function buildInteractiveResult(
       return;
     }
     wasOpened = true;
-    // TODO: Log on click
+    engine.dispatch(
+      productClick({
+        // TODO: If we used the relay Product, we would not need to construct the product here.
+        product: {
+          productId: props.options.product.permanentid,
+          name: props.options.product.ec_name ?? '',
+          price: props.options.product.ec_price ?? 0,
+        },
```

Comment:
Agreed with the todo. A "benefit" of the current approach is it by-passes the typescript check, which means one would only realize the prices are always 0 at runtime, hmm.